### ISSUE TITLE
Serialization does not work

### ISSUE THREAD

**Issue by RasterBurn** [greglook/clj-pgp#9]
```clojure
(def keypair (pgp-gen/generate-keypair (pgp-gen/rsa-keypair-generator 2048) :rsa-general))
(def pubkey (.getPublicKey keypair))
(let [ascii (pgp/encode-ascii pubkey)]
  (print ascii)
  (= ascii (pgp/encode-ascii (pgp/decode-public-key ascii))))
```

Here's what the `print` produced:
```
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: BCPG v1.51

mQENBFT3ek8BCADPJF34pFhiMjtMugYPD7lIe0VkyQQ5NBhVe00PhCP8RmmjmiiE
j+r7mAIXV8hX25A11U/M/QFGXztVcR7HaCmLAWHpaYbgAYLFHnRpUgyLRf4VvdTk
62FAJDLv865NBegW+49BwMI0HihNJiKBsMb/UULvpgOa4seRqE+EeUqwDp3AmxcS
P1GipWXXibjnJFfzbqVqoi27dsGwa443U9vZUSCufaLiH12BpIPYlnfeVCT4H5iU
EXrisJhdWk4Zge/3E5CiGNdwgvUQUfZEjIw4EeExMZabysAR5tktaNBi63G1c0fS
r/usODTQUwIqKK8ksidMPcqBLxZgJHXddd+1ABEBAAE=
=X76Z
-----END PGP PUBLIC KEY BLOCK-----
```

And here's the error:
```
IllegalStateException Data did not contain a public key: org.bouncycastle.openpgp.PGPPublicKeyRing@3feb6c8e clj-pgp.core/decode-public-key (core.clj:264)
```

This is clj-pgp 0.8.0
bouncycastle 1.51

**Comment by RasterBurn** [comment]
I tried the following:

```clojure
(print (pgp/decode ascii))
```

and got back:

```clojure
(#<PGPPublicKeyRing org.bouncycastle.openpgp.PGPPublicKeyRing@7277b4c6>)
```

I believe `decode-public-key` checks for type `PGPPublicKey`, not `PGPPublicKeyRing`.

**Comment by RasterBurn** [comment]
This seems to work:
```clojure
(.getPublicKey (first (pgp/decode (pgp/encode-ascii pubkey))))

#<PGPPublicKey org.bouncycastle.openpgp.PGPPublicKey@5fd19525>
```

**Comment by greglook** [comment]
Ah, thanks for the report! I thought I'd covered this with a test case, but I'll investigate.

**Comment by greglook** [comment]
Wait, I actually [do have a test case](https://github.com/greglook/clj-pgp/blob/5570627a923cc20fe5a7d2ae9b43e3165e3ad51f/test/clj_pgp/test/key_utils.clj#L148) for this! I'm not sure why that doesn't work for you.

**Comment by greglook** [comment]
@RasterBurn see if the latest commit in develop (6524fb7160fc52d92b76788662938359136dc58a) fixes your issue.

**Comment by RasterBurn** [comment]
Yes, this fixes the issue!   Thanks!

**Comment by greglook** [comment]
Released [0.8.1](https://clojars.org/mvxcvi/clj-pgp/versions/0.8.1) which includes this fix.
